library/std/src/env.rs

```diff
@@ -338,10 +338,41 @@ impl Error for VarError {
 /// ```
 #[stable(feature = "env", since = "1.0.0")]
 pub fn set_var<K: AsRef<OsStr>, V: AsRef<OsStr>>(key: K, value: V) {
+    // Safety: This isn't sound. See #27970 for details.
+    unsafe { _set_var(key.as_ref(), value.as_ref()) }
+}
+
+/// Sets the environment variable `key` to the value `value` for the currently running
+/// process.
+///
+/// # Panics
+///
+/// This function may panic if `key` is empty, contains an ASCII equals sign `'='`
+/// or the NUL character `'\0'`, or when `value` contains the NUL character.
+///
+/// # Safety
+///
+/// Some platforms only expose non-threadsafe APIs for altering the environment.
+/// On these platforms, you must ensure this method is not called when the
+/// program is multithreaded.
```

Comment:
```suggestion
/// program is multithreaded and any other thread could be reading or writing environment.
```

If you actually control all the code in other threads, and *know* it can't touch the environment, this is still safe.